src/scripts/main.js

```diff
@@ -1,3 +1,22 @@
 'use strict';
 
-// write code here
+const employeeList = document.querySelector('ul');
+
+function sortList(listContainer) {
+  const listElements = Array.from(listContainer.children);
+
+  listElements.sort((a, b) => {
+    const salaryA = parseSalary(a.dataset.salary);
+    const salaryB = parseSalary(b.dataset.salary);
+
+    return salaryB - salaryA;
+  });
+
+  listElements.forEach((item) => listContainer.appendChild(item));
+}
+
+function parseSalary(salaryStr) {
+  return Number(salaryStr.replace(/[^0-9.-]+/g, ''));
```

Comment:
The `parseSalary` function uses a regular expression to remove non-numeric characters from the salary string. While this works for typical salary formats, it might not handle cases where the salary string contains unexpected characters or formats (e.g., currency symbols in different positions). Ensure that the input format is consistent or consider adding validation to handle such cases.